server/src/logic/handler/routes/recipe.handler.ts

```diff
@@ -17,6 +20,60 @@ export class RecipeHandler {
       .catch((e: any) => res.status(e.statusCode).json(e.forResponse()));
   }
 
+  public getAll = (req: Request, res: Response): void => {
+
+    const queryPramas = req.query;
+    const allQueryParams = Object.getOwnPropertyNames(queryPramas);
+    const firstQeryParam = allQueryParams[0];
+    const valueFirstQueryParam = queryPramas[firstQeryParam];
+
+    if (firstQeryParam && valueFirstQueryParam) {
+      if (!(new Recipe().fields.has(firstQeryParam))) {
+        const e = new ApiError(ERRORS.RECIPE_INVALID_FIELD.withArgs(firstQeryParam))
+          .addSource(RecipeHandler.name)
+          .addSourceMethod("getAll()");
+        res.status(e.statusCode).json(e.forResponse());
+      }
+      this._db.recipes.getByField(firstQeryParam, valueFirstQueryParam)
+      .then((result: Recipe[]) => res.status(200).json(result))
+      .catch((e: any) => res.status(e.statusCode).json(e.forResponse()));
+    } else {
+      this._db.recipes.getAll()
+      .then((result: Recipe[]) => res.status(200).json(result))
+      .catch((e: any) => {
+        if (e.statusCode === 404) {
```

Comment:
I will compare the error code (from the ERRORS class), I think this will be more specific.